_posts/2021-02-04-painless-code-formatting-with-editor-config.md

```diff
@@ -0,0 +1,84 @@
+
+Since there are lot of IDEs and we cannot touch all of them, I have selected two IDEs: IntelliJ IDEA which comes with native support for EditorConfig and Eclipse which requires a plugin to be downloaded for it to work properly. 
+
+For a complete list of supported IDEs (those requiring plugin and those with native support), please check the [official website](https://editorconfig.org/).
+
+## Using editorconfig with IntelliJ
+
+IntelliJ comes with native support for EditorConfig, which means that we do not have to install a plugin to make it work. To get started, you need to create a file named .editorconfig in the root folder of your project and define the coding styles you need.
+
+Since I would like my java code to use tab indentation with a tab size of 4, utf-8 character set and trim any trailing whitespaces in my code. I will define the following properties in the .editorconfig file to achieve this:
+
+```
+# Topmost editor config file
+root = true
+
+# Custom Codng Styles for Java files
+[*.java]
+
+# The other allowed value you can use is space
+indent_style = tab 
+
+# You can play with this value and set it to how big you want your tab to be
+indent_size = 4
+
+# Character set to be used in java files.
+charset = utf-8 
+trim_trailing_whitespace = true
+```
+
+In the snippet above, I have defined two major sections: root definition section and the Java style section. 
+
+I have specified the **root** value to be *true* and what this means is that when a file is opened, editorConfig plugin will start searching for .editorconfig files starting from the current directory. The search will only stop when it has reached the root directory or when it sees a .editorconfig file with root value set to true.
+
+EditorConfig plugin applies styles in a top down fashion, so if you have several .editorconfig files in your project with some duplicated properties, the closet .editorconfig file takes precedence.
+
+For the Java section, I have defined a pattern **[*.java]** to match all java files. If your requirement is to match some other type of files with a different extension, a complete list of Wildcard Pattern is available in the [official website](https://editorconfig.org/)
+
+You can use the Editor and Preview button (this requires that you create at least a Java class in your project) at the top-right corner of IntelliJ to view the effect of the styles in realtime. To apply the styles to all Java classes in your project, you would right click on the project and select *Reformat Code* from the list of options, a dialog box should apply, and you can click on the *Run* button to apply your style changes. Once done, you should see all your Java source files neatly formatted according to the styles you have defined in .editorconfig file.
```

Comment:
I can't see the button you mean. Can you elaborate on that, or perhaps add an image of that button? And perhaps an image of the dialog?

Reply:
done @thombergs